core/src/main/java/uk/gov/justice/services/core/jms/converter/EnvelopeConverter.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.services.core.jms.converter;
+
+import uk.gov.justice.services.core.jms.exception.JmsConverterException;
+import uk.gov.justice.services.core.util.JsonObjectConverter;
+import uk.gov.justice.services.messaging.Envelope;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+/**
+ * Implementation of {@link MessageConverter} for {@link uk.gov.justice.services.messaging.Envelope}
+ */
+@ApplicationScoped
+public class EnvelopeConverter implements MessageConverter<Envelope, TextMessage> {
+
+    static final String JMS_HEADER_CPPNAME = "CPPNAME";
+
+    @Inject
+    JsonObjectConverter jsonObjectConverter;
+
+    @Override
+    public Envelope fromMessage(final TextMessage message) {
+        String messageAsString = null;
+
+        try {
+            messageAsString = message.getText();
+            return jsonObjectConverter.asEnvelope(jsonObjectConverter.fromString(messageAsString));
+        } catch (JMSException e) {
+            try {
+                throw new JmsConverterException(String.format("Exception while creating envelope from message %s", message.getJMSMessageID()), e);
```

Comment:
The String "Exception while creating envelope from message" could be a constant as it is repeated.